proposals/lip-0050.md

```diff
@@ -213,6 +213,42 @@ getLegacyAccount(publicKey)
                 "balance": balance}
 ```
 
+### Genesis Block Processing
+
+#### Genesis State Initialization
+
+Let `genesisBlockAssetBytes` be the `data` bytes included in the block assets for the legacy module and let `genesisBlockAssetObject` be the deserialization of `genesisBlockAssetBytes` according to the `genesisLegacyStoreSchema` schema given below. If the deserialization fails, reject the block.
+
+```json
+genesisLegacyStoreSchema = {
+    "type": "object",
+    "required": ["accounts"],
+    "properties": {
+        "accounts": {
+            "type": "array",
+            "fieldNumber": 1,
+            "items": {
+                "type": "object",
+                "required": ["address", "balance"],
+                "properties": {
+                    "address": {
+                        "dataType": "bytes",
+                        "fieldNumber": 1
+                    },
+                    "balance": {
+                        "dataType": "uint64",
+                        "fieldNumber": 2
+                    }
+                }
+            }
+        }
+    }
+}
+```
+Then, for every entry `account` in `genesisBlockAssetObject.accounts`:
```

Comment:
We could also add the check that `address` values must be unique. 
And that the array is sorted by lexicographical order of address. This last one is not super important imo, but we have been doing it in other similar situations.

Reply:
I added the uniqueness check. 
Regarding the order: I have a small preference to add this requirement to the mirgration LIP because there the need for it (all computed snapshot blocks must be the same) is more obvious. But if you find it important to have it here, I'll change it.